### PR TITLE
Promote the Change Management WD to PSD03

### LINKED ISSUE #96: CM Actions file should be linked or removed
What is oslc-specs/specs/cm/change-mgt-actions.html doing in the repo?

### PR DESCRIPTION
**Pull request by jamsden**
This will be our first oslc-op published document. Once the pull request is approved and merged, it should be tagged with cm-v3.0-psd01. At that point I can render the HTML and provide to OASIS for publishing.

Closes  #96

### CODE REVIEW

File: specs/cm/change-mgt-spec.html [hunk 1]
```diff
@@ -36,8 +36,8 @@
 
       // Other parts of multi-part spec
       additionalArtifacts: [
-        { title: "OSLC Change Management Version 3.0. Part 1: Specification (this document)", href: thisBase + filePath },
-        { title: "OSLC Change Management Version 3.0. Part 2: Vocabulary", href: thisBase + "/change-mgt-vocab.html" },
+        { title: "OSLC Change Management Version 3.0. Part 1: Specification (this document)", href:  "change-mgt-spec.html" },
```
Comment (berezovskyi) [inline specs/cm/change-mgt-spec.html]: `thisBase` produces absolute URIs, but this might work just fine too!

**Comment by berezovskyi** [comment]
@ndjc do you think we can publish this or do we need to wait for a fix in https://github.com/oasis-tcs/tab-respec/issues/8#issuecomment-520551140 (or add a boilerplate conformance section?)

**Comment by berezovskyi** [comment]
@jamsden please pull the changes to the branch and check that it renders for you too!

**Comment by berezovskyi** [comment]
- [ ] @berezovskyi to set the https://open-services.net/spec/cm/latest and https://open-services.net/spec/cm/latest-draft redirects properly

**Comment by jamsden** [comment]
looks ok to me. Notice that the additionalArtifacts can't use the filePath variable because it has a / in it. There appears to be no case where filePath is used without additional path prefixes. Suggest renaming this variable to fileName, and put the / at the end of wdBase and oasisBase. That helps one know that the end path component is a path, not a file.

**Comment by berezovskyi** [comment]
Filename is good, that’s exactly what I did in Core after you told me. Look at its source for an example.

We can merge this anyway because we don’t need that variable in additionalArtifacts.

File: specs/cm/change-mgt-actions.html [hunk 2]
```diff
@@ -64,7 +64,6 @@
       // This is important for Rec-track documents, do not copy a patent URI from a random
       // document unless you know what you're doing. If in doubt ask your friendly neighbourhood
       // Team Contact.
-      wgPatentURI:  "",
```
Comment (berezovskyi) [inline specs/cm/change-mgt-actions.html]: we got to move this file out of this folder before merging

**Comment by jamsden** [comment]
I don't see why leaving content in the folder is a problem. No matter where we put it in the repo its going to get a release tag. And this is certainly something we want to preserve. What problem does leaving it cause? No published document will reference it.

**Comment by berezovskyi** [comment]
> I don't see why leaving content in the folder is a problem. No matter where we put it in the repo its going to get a release tag. And this is certainly something we want to preserve. What problem does leaving it cause? 

For me it's more of a signal we are sending. Me and you _know_ that CM Actions are not part of CM PSD03 but someone outside might not. They will most likely assume that the whole specs/cm/ folder is PSD03. I don't care where you move it, but please move it to the place of your liking.

**Comment by berezovskyi** [comment]
https://open-services.net/spec/cm/latest and https://open-services.net/spec/cm/latest-draft are now working properly

**Comment by berezovskyi** [comment]
https://github.com/oslc-op/oslc-specs/releases/tag/untagged-edbf0174cbb8a788e423 is the draft release with the zipped up snapshots

**Comment by jamsden** [comment]
A bit of history: Actions use to be a separate specification and TC, lead by Martin Pain. This TC was abandoned because it had not met in over six months and didn't produce any documents. 

We decided to move the actions content into core because there were previous documents in open-services.net, including automation, that we felt needed to be preserved, and that might become OASIS standards. However, the demand for these specifications did not materialize and so no additional work was done.

The actions.html spec in core is very different then the change-mgt-actions.html in cm. The core document describes a generic mechanism for defining and executing actions for any purpose. The cm actions document is a very carefully crafted document that extends core actions with an action that can specify a ChangeRequest target state. 

By carefully crafted I mean that CM actions was a compatible extension to core actions that didn't require any of the complex stuff in automation and actions that we didn't necessarily want to take forward. 

But then we decided not to include ways to change the state of a ChangeRequest with a POST - just use a PUT or PATCH and let the server decide if the value you specify is valid.

So these specs should be kept as we basically deferred these capabilities in core and cm. I also think they are where they should be as the represent possible future work.

**Comment by berezovskyi** [comment]
Wow! Thanks for all this info. Good context.

What do you mean “keep because they were deferred”? I suggest we keep those files but not in specs/core and not specs/cm if they are not becoming part of the PS.

--
/Andrew
(from phone)

13 aug. 2019 kl. 22:12 skrev Jim Amsden <notifications@github.com<mailto:notifications@github.com>>:


A bit of history: Actions use to be a separate specification and TC, lead by Martin Pain. This TC was abandoned because it had not met in over six months and didn't produce any documents.

We decided to move the actions content into core because there were previous documents in open-services.net<http://open-services.net>, including automation, that we felt needed to be preserved, and that might become OASIS standards. However, the demand for these specifications did not materialize and so no additional work was done.

The actions.html spec in core is very different then the change-mgt-actions.html in cm. The core document describes a generic mechanism for defining and executing actions for any purpose. The cm actions document is a very carefully crafted document that extends core actions with an action that can specify a ChangeRequest target state.

By carefully crafted I mean that CM actions was a compatible extension to core actions that didn't require any of the complex stuff in automation and actions that we didn't necessarily want to take forward.

But then we decided not to include ways to change the state of a ChangeRequest with a POST - just use a PUT or PATCH and let the server decide if the value you specify is valid.

So these specs should be kept as we basically deferred these capabilities in core and cm. I also think they are where they should be as the represent possible future work.

**Comment by jamsden** [comment]
By deferred, I mean these might become active in the future. Hence the reason for leaving them where they are.

**Comment by berezovskyi** [comment]
I appreciate the consideration for the future development of the spec. However, I find it hard to understand how including a file in specs/cm/ and tagging the commit indicating CM PSD03 revision wouldn’t create a false impression of actions being part of PSD03. Furthermore, as long as it stays there, it gets rendered under https://oslc-op.github.io/oslc-specs/specs/cm/change-mgt-actions.html

You even set the status on actions to PSD and unified the naming of the files: https://github.com/oslc-op/oslc-specs/commit/42ed1a03fe9a53aa93d38601d579dc00604077d2 How do you not find it to be confusing even a slightest bit?

If you insist keeping the CM where it is, I can only suggest removing the file on the master branch, then creating another branch, say, cm-actions, and adding that file there.

**Comment by jamsden** [comment]
Its not part of the multi-part specification, and only people rooting around in the repo would see it.

**Pull request merged**